Gem/Code/Source/Components/NetworkWeaponsComponent.cpp

```diff
@@ -91,24 +91,10 @@ namespace MultiplayerSample
 
     void NetworkWeaponsComponent::ActivateWeaponWithParams(WeaponIndex weaponIndex, WeaponState& weaponState, const FireParams& fireParams, bool validateActivations)
     {
-        const uint32_t weaponIndexInt = aznumeric_cast<uint32_t>(weaponIndex);
```

Comment:
Not necessarily something to do in this change, but we might want a cvar that caps the max positional and orientational delta from current bone state between the input and the animation system. If we exceed the tolerance we can just throw away the input value and use the animation state.

Reply:
Yeah I intend to do just that before I pull this out of draft. A lot of the Weapon firing logic in general needs a good look.